pycbc/cosmology.py

```diff
@@ -23,88 +23,228 @@
 #
 """
 This modules provides functions for computing cosmological quantities, such as
-redshift.
+redshift. This is mostly a wrapper around ``astropy.cosmology``.
+
+Note: in all functions, ``distance`` is short hand for ``luminosity_distance``.
+Any other distance measure is explicitly named; e.g., ``comoving_distance``.
 """
 
+import logging
 import numpy
-import lal
 from scipy import interpolate
+import astropy.cosmology
+from astropy import units
+from astropy.cosmology.core import CosmologyError
 from pycbc.conversions import ensurearray, formatreturn
 
-class _DistToZ(object):
-    """Class to convert luminosity distance to redshift using the given
-    cosmology.
+DEFAULT_COSMOLOGY = 'Planck15'
+
+
+def get_cosmology(cosmology=None, **kwargs):
+    r"""Gets an astropy cosmology class.
+
+    Parameters
+    ----------
+    cosmology : str or astropy.cosmology.FlatLambdaCDM, optional
+        The name of the cosmology to use. For the list of options, see
+        :py:attr:`astropy.cosmology.parameters.available`. If None, and no
+        other keyword arguments are provided, will default to
+        :py:attr:`DEFAULT_COSMOLOGY`. If an instance of
+        :py:class:`astropy.cosmology.FlatLambdaCDM`, will just return that.
+    \**kwargs :
+        If any other keyword arguments are provided they will be passed to
+        :py:attr:`astropy.cosmology.FlatLambdaCDM` to create a custom
+        cosmology.
+
+    Returns
+    -------
+    astropy.cosmology.FlatLambdaCDM
+        The cosmology to use.
+
+    Examples
+    --------
+    Use the default:
+
+    >>> from pycbc.cosmology import get_cosmology
+    >>> get_cosmology()
+    FlatLambdaCDM(name="Planck15", H0=67.7 km / (Mpc s), Om0=0.307,
+                  Tcmb0=2.725 K, Neff=3.05, m_nu=[0.   0.   0.06] eV,
+                  Ob0=0.0486)
+
+    Don't trust Planck? Use WMAP!
+
+    >>> get_cosmology("WMAP9")
+    FlatLambdaCDM(name="WMAP9", H0=69.3 km / (Mpc s), Om0=0.286, Tcmb0=2.725 K,
+                  Neff=3.04, m_nu=[0. 0. 0.] eV, Ob0=0.0463)
+
+    Don't trust anyone? Create your own!
```

Comment:
maybe a bit less chirpy dialogue and a bit more explanation of what happens when you specify some cosmology parameters yourself but not others .. presumably astropy then chooses for you, which is what results in Neff = 3.04 coming out of 'nowhere' ?

Reply:
Ok, I modified it to be less cheeky. I also pointed the user to the docs for `FlatLambdaCDM` to see where the defaults come from.